test/units/testsuite-68.sh

```diff
@@ -0,0 +1,154 @@
+[Unit]
+Description=TEST-68-PROPAGATE-EXIT-STATUS with OnSuccess= trigger
+OnSuccess=testservice-success-exit-handler-68.service
+
+[Service]
+ExecStart=/bin/bash -c "exit 0"
+EOF
+
+# Script to check that when an OnSuccess= dependency fires, the correct
+# MONITOR* env variables are passed.
+cat >/tmp/check_on_success.sh <<EOF
+#!/usr/bin/env bash
+
+set -ex
+
+if [ "\$MONITOR_SERVICE_RESULT" != "success" ]; then
+    echo 'MONITOR_SERVICE_RESULT was "\$MONITOR_SERVICE_RESULT", expected "success"';
+    exit 1;
+fi
+
+if [ "\$MONITOR_EXIT_CODE" != "exited" ]; then
+    echo 'MONITOR_EXIT_CODE was "\$MONITOR_EXIT_CODE", expected "exited"';
+    exit 1;
+fi
+
+if [ "\$MONITOR_EXIT_STATUS" != "0" ]; then
+    echo 'MONITOR_EXIT_STATUS was "\$MONITOR_EXIT_STATUS", expected "0"';
+    exit 1;
+fi
+
+if [ -z "\$MONITOR_INVOCATION_ID" ]; then
+    echo 'MONITOR_INVOCATION_ID unset';
+    exit 1;
+fi
+
+if [ "\$MONITOR_UNIT" != "testservice-success-68.service" ]; then
+    echo 'MONITOR_UNIT was "\$MONITOR_UNIT", expected "testservice-success-68.service"';
+    exit 1;
+fi
+
+exit 0;
+EOF
+chmod +x /tmp/check_on_success.sh
+
+# Handle testservice-failure-exit-handler-68.service exiting with success.
+cat >/run/systemd/system/testservice-success-exit-handler-68.service <<EOF
+[Unit]
+Description=TEST-68-PROPAGATE-EXIT-STATUS handle service exiting in success
+
+[Service]
+ExecStartPre=/tmp/check_on_success.sh
+ExecStart=/tmp/check_on_success.sh
+EOF
+
+# Script to check that when an OnFailure= dependency fires, the correct
+# MONITOR* env variables are passed.
+cat >/tmp/check_on_failure.sh <<EOF
+#!/usr/bin/env bash
+
+set -ex
+
+if [ "\$MONITOR_SERVICE_RESULT" != "exit-code" ]; then
+    echo 'MONITOR_SERVICE_RESULT was "\$MONITOR_SERVICE_RESULT", expected "success"';
+    exit 1;
+fi
+
+if [ "\$MONITOR_EXIT_CODE" != "exited" ]; then
+    echo 'MONITOR_EXIT_CODE was "\$MONITOR_EXIT_CODE", expected "exited"';
+    exit 1;
+fi
+
+if [ "\$MONITOR_EXIT_STATUS" != "1" ]; then
+    echo 'MONITOR_EXIT_STATUS was "\$MONITOR_EXIT_STATUS", expected "1"';
+    exit 1;
+fi
+
+if [ -z "\$MONITOR_INVOCATION_ID" ]; then
+    echo 'MONITOR_INVOCATION_ID unset';
+    exit 1;
+fi
+
+if [ "\$MONITOR_UNIT" != "testservice-failure-68.service" ]; then
+    echo 'MONITOR_UNIT was "\$MONITOR_UNIT", expected "testservice-failure-68.service"';
+    exit 1;
+fi
+
+exit 0;
+EOF
+chmod +x /tmp/check_on_failure.sh
+
+
+# Handle testservice-failure-exit-handler-68.service exiting with failure.
+cat >/run/systemd/system/testservice-failure-exit-handler-68.service <<EOF
+[Unit]
+Description=TEST-68-PROPAGATE-EXIT-STATUS handle service exiting in failure
+
+[Service]
+ExecStartPre=/tmp/check_on_failure.sh
+ExecStart=/tmp/check_on_failure.sh
+EOF
+
+systemctl daemon-reload
+
+systemctl start testservice-failure-68.service
+systemctl start testservice-success-68.service
+# We expect both services to exit without failure.
+sleep 5
```

Comment:
The sleep here really isn't required in hindsight since the above commands will not return until the services have been started and have spawned their OnFailure=/OnSuccess= dependencies.

What I am wanting to do with this test is:

1) Start some services which have OnFailure= or OnSuccess= dependencies.
2) When those dependencies fire, I want to check that the correct monitor vars are set in the triggered service.

I guess there may be a race where the ```wait_on_state_or_fail "testservice-failure-exit-handler-68.service" "inactive" "10"``` returns immediately since the state of the unit before it is activated is 'inactive' already.